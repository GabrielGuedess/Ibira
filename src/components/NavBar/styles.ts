import styled, { css } from 'styled-components';

export const Wrapper = styled.menu`
  ${({ theme }) => css`
    width: 100%;
    height: 9rem;
    padding: 0.8rem 1.6rem;
    background: ${theme.colors.primary};
  `}
`;
