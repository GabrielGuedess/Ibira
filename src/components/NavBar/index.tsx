import * as S from './styles';

const NavBar = () => (
  <S.Wrapper>
    <h1>NavBar</h1>
  </S.Wrapper>
);

export default NavBar;
