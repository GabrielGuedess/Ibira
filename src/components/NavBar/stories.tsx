import { Story, Meta } from '@storybook/react/types-6-0';
import NavBar, { MenuProps } from '.';

export default {
  title: 'NavBar',
  component: NavBar,
} as Meta;

export const Desktop: Story<MenuProps> = args => <NavBar {...args} />;

Desktop.parameters = {
  layout: 'fullscreen',
};

export const Mobile: Story<MenuProps> = args => <NavBar {...args} />;

Mobile.parameters = {
  layout: 'fullscreen',
  viewport: {
    defaultViewport: 'mobile1',
  },
};
