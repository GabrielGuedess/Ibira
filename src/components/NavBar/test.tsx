import { render, screen } from '@testing-library/react';

import NavBar from '.';

describe('<NavBar />', () => {
  it('should render the heading', () => {
    const { container } = render(<NavBar />);

    expect(
      screen.getByRole('heading', { name: /NavBar/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
