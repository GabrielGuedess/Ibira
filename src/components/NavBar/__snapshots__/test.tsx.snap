// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    NavBar
  </h1>
</main>
`;
